index.js

```diff
@@ -146,6 +196,32 @@ class SocketValidationError extends Error {
   }
 }
 
+/**
+ * @param {object} parsedV1Content
+ * @returns {Promise<SocketYml | null>}
+ */
+async function parseV1SocketConfig (parsedV1Content) {
+  if (!validateV1(parsedV1Content)) {
+    return null
+  }
+
+  const v2 = {
+    version: 2,
+    projectIgnorePaths: parsedV1Content?.ignore ?? [],
+    issueRules: parsedV1Content?.issues ?? {},
+    githubApp: {
+      enabled: parsedV1Content?.enabled,
+      pullRequestAlertsEnabled: parsedV1Content?.pullRequestAlertsEnabled,
+      projectReportsEnabled: parsedV1Content?.projectReportsEnabled
+    }
+  }
+
+  if (!validate(v2)) {
+    return null
+  }
+  return v2
```

Comment:
```suggestion
  /** @type {SocketYml} */
  const v2 = {
    version: 2,
    projectIgnorePaths: parsedV1Content?.ignore ?? [],
    issueRules: parsedV1Content?.issues ?? {},
    githubApp: {
      enabled: parsedV1Content?.enabled,
      pullRequestAlertsEnabled: parsedV1Content?.pullRequestAlertsEnabled,
      projectReportsEnabled: parsedV1Content?.projectReportsEnabled
    }
  }

  return v2
```

How about typing it up right away? As we know the type of `parsedV1Content` we can then know through the type system that the `SocketYml` will be valid as well?